pkg/action/archive.go

```diff
@@ -0,0 +1,225 @@
+package action
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ulikunitz/xz"
+)
+
+const maxBytes = 1 << 29 // 512MB
+
+// copyArchive copies the source archive file to the temporary directory.
+func copyArchive(src string, dst string) error {
```

Comment:
We might want to plumb context here so that we get the correct logger.

Reply:
Added in [`424a907` (#144)](https://github.com/chainguard-dev/bincapz/pull/144/commits/424a907f4bbc7201f6f4bc242c2e2b0ac06ec853).